src/main/kotlin/app/revanced/patches/shared/settings/preference/BasePreference.kt

```diff
@@ -13,6 +13,7 @@ import org.w3c.dom.Element
 internal abstract class BasePreference(
     override val key: String,
     override val title: StringResource,
+    val summary: StringResource? = null,
```

Comment:
Each preference seems to have a `summary`. Thus, it should be present in `IPreference` whereas it should be serialized in this classes implementation of the method `serialize`. Each preference either extends from `BasePreference` or `IPreference`, thus the declaration of `serialize` should be present in `IPreference`

Reply:
```kotlin

internal interface IPreference {
    /**
     * Key of the preference.
     */
    val key: String

    /**
     * Title of the preference.
     */
    val title: StringResource

    /**
     * Tag name of the preference.
     */
    val tag: String

    /**
     * Summary of the preference.
     */
    val summary: StringResource?

    /**
     * Serialize preference element to XML.
     *
     * @param ownerDocument Target document to create elements from.
     * @param resourceCallback Called when a resource has been processed.
     * @return The serialized element.
     */
    fun serialize(ownerDocument: Document, resourceCallback: ((IResource) -> Unit)? = null): Element
}

```